src/human-readable/types/signatures.ts

```diff
@@ -128,7 +129,7 @@ export type IsName<TName extends string> = TName extends ''
   : false
 export type ValidateName<
   TName extends string,
-  CheckCharacters extends boolean = false,
+  CheckCharacters extends boolean = ResolvedConfig['Strict'],
```

Comment:
This now gets enabled once folks enable strict mode instead of always being false.